### PR TITLE
feat(connectors, soap): added SOAP connector doc

### PR DESCRIPTION
**Pull request by igpetrov**
## Description

feat(connectors, soap): added SOAP connector doc

Ready to be merged once @crobbins215 confirms that the doc is sufficient. The feature has been release in 8.5 as close-beta for several users. Now it's ready to go public.

Only for self-managed!

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer.

### CODE REVIEW

File: versioned_docs/version-8.5/components/connectors/protocol/soap.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+id: soap
+title: SOAP Connector
+sidebar_label: SOAP Connector
+description: The SOAP Connector allows you to connect your BPMN with SOAP services.
+---
+
+**SOAP** is a messaging protocol specification for exchanging structured
+information in the implementation of web services in computer networks.
+
+The **SOAP Connector** allows you to interact with [SOAP](https://en.wikipedia.org/wiki/SOAP) services endpoints
```
Comment (crobbins215) [inline versioned_docs/version-8.5/components/connectors/protocol/soap.md:11]: "services endpoints" should be "service endpoints"?

File: versioned_docs/version-8.5/components/connectors/protocol/soap.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+## Prerequisites
+
+The **SOAP Connector** is only supported by self-managed Camunda instances.
+
+To use the **SOAP Connector**, you need to have an active SOAP service.
+
+## Create a SOAP Connector task
+
+To use the **SOAP Connector** in your process, either change the type of existing task by clicking on it and using
```
Comment (crobbins215) [inline versioned_docs/version-8.5/components/connectors/protocol/soap.md:22]: "either change the type of existing task" should be "either change the type of an existing task"?

File: versioned_docs/version-8.5/components/connectors/protocol/soap.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+the wrench-shaped **Change type** context menu icon, or create a new Connector task by using the **Append Connector** context menu.
+Follow our [guide to using Connectors](/components/connectors/use-connectors/index.md) to learn more.
+
+## Connection
+
+Enter your SOAP service URL in the field **Service URL**, for example `https://myservice.com/service/MyService.wso`.
+
+## Authentication
+
+Select authentication type from the **Authentication** dropdown.
```
Comment (crobbins215) [inline versioned_docs/version-8.5/components/connectors/protocol/soap.md:32]: "Select authentication type from the" should be "Select the authentication type from the"

File: versioned_docs/version-8.5/components/connectors/protocol/soap.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+You have to enter the **Namespaces** value, e.g.
+`={"camunda":"http://my.service.com/webservicesserver/"}`.
+
+## Output mapping
+
+### Result variable
+
+You can export a complete response from an SOAP call into a dedicated variable accessible anywhere in a process.
```
Comment (crobbins215) [inline versioned_docs/version-8.5/components/connectors/protocol/soap.md:101]: "You can export a complete response from an SOAP call" should be "You can export a complete response from a SOAP call"

File: docs/components/connectors/protocol/soap.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+You can export a complete response from an SOAP call into a dedicated variable accessible anywhere in a process.
+To do so, just input a variable name in the **Result variable** field. We recommend using a unique name to avoid
+variables being overwritten.
+
+## Usage examples
+
+### Example 1
+
+For example, you would like to send a following SOAP request:
```
Comment (crobbins215) [inline docs/components/connectors/protocol/soap.md:109]: "a following SOAP request" should be "the following SOAP request"

File: docs/components/connectors/protocol/soap.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+URL: `https://myservice:8888/webservice.wso`
+
+Body:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <Object01 mlns="http://www.my.namespace.com/namespace/">
```
Comment (crobbins215) [inline docs/components/connectors/protocol/soap.md:148]: Should this be xmlns or mlns?

File: versioned_docs/version-8.5/components/connectors/protocol/soap.md [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+URL: `https://myservice:8888/webservice.wso`
+
+Body:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <Object01 mlns="http://www.my.namespace.com/namespace/">
```
Comment (crobbins215) [inline versioned_docs/version-8.5/components/connectors/protocol/soap.md:148]: Same for this, is it mlns or xmlns? Im not a soap expert but flagging just incase

File: versioned_docs/version-8.5/components/connectors/protocol/soap.md [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+
+```json
+{
+  "ns:Object01": {
+    "ns:Object02": 12345
+  }
+}
+```
+
+Please, pay attention, that here we introduced a new `ns:` prefix. Prefix, can be any arbitrary string, that is not defined as namespace.
```
Comment (crobbins215) [inline versioned_docs/version-8.5/components/connectors/protocol/soap.md:165]: I would reword this "Prefix, can be any arbitrary string, that is not defined as namespace." to be "The prefix can be any arbitrary string that is not defined as a namespace."

File: docs/components/connectors/protocol/soap.md [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+
+```json
+{
+  "ns:Object01": {
+    "ns:Object02": 12345
+  }
+}
+```
+
+Please, pay attention, that here we introduced a new `ns:` prefix. Prefix, can be any arbitrary string, that is not defined as namespace.
```
Comment (crobbins215) [inline docs/components/connectors/protocol/soap.md:165]: Use softer language, something like "Please note that here we introduced a new ns: prefix."

File: docs/components/connectors/protocol/soap.md [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <Object01>
+      <Object02>12345</Object02>
+    </Object01>
+  </soap:Body>
+</soap:Envelope>
+```
+
+For that, set the **SOAP body** dropdown to **Template**.
```
Comment (crobbins215) [inline docs/components/connectors/protocol/soap.md:195]: "For that, set the SOAP body dropdown to Template." could be "To do this, set the SOAP body dropdown to Template."

**Review by crobbins215 — COMMENTED** [review]
Made some suggestions to review @igpetrov, please take a look.

**Comment by igpetrov** [comment]
> Made some suggestions to review @igpetrov, please take a look.

@crobbins215 good point. Since I am not a native speaker, this is where @christinaausley usually helps us. I will update my PR with the comments suggested. Thank you!

**Comment by crobbins215** [comment]
> > Made some suggestions to review @igpetrov, please take a look.
> 
> @crobbins215 good point. Since I am not a native speaker, this is where @christinaausley usually helps us. I will update my PR with the comments suggested. Thank you!

Yeah no problems at all, good doc. Thanks @igpetrov.

@christinaausley if you have time, can check over it for any of my Britishisms that might sneak in :)

File: versioned_docs/version-8.5/components/connectors/protocol/soap.md (hunk 2 above)
Comment (igpetrov) [inline versioned_docs/version-8.5/components/connectors/protocol/soap.md:22]:
```suggestion
To use the **SOAP Connector** in your process, either change the type of an existing task by clicking on it and using
```

File: docs/components/connectors/protocol/soap.md [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+
+## Prerequisites
+
+The **SOAP Connector** is only supported by self-managed Camunda instances.
+
+To use the **SOAP Connector**, you need to have an active SOAP service.
+
+## Create a SOAP Connector task
+
+To use the **SOAP Connector** in your process, either change the type of existing task by clicking on it and using
```
Comment (igpetrov) [inline docs/components/connectors/protocol/soap.md:22]:
```suggestion
To use the **SOAP Connector** in your process, either change the type of an existing task by clicking on it and using
```

File: versioned_docs/version-8.5/components/connectors/protocol/soap.md (hunk 3 above)
Comment (igpetrov) [inline versioned_docs/version-8.5/components/connectors/protocol/soap.md:32]:
```suggestion
Select the authentication type from the **Authentication** dropdown.
```

**Comment by igpetrov** [comment]
Suggestions updated

File: docs/components/connectors/protocol/soap.md [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+---
+
+:::note
+The **SOAP Connector** is only supported by Self-Managed Camunda 8 instances.
+:::
+
+**Simple Object Access Protocol (SOAP)** is a messaging protocol specification for exchanging structured
+information in the implementation of web services in computer networks.
+
+The **SOAP Connector** allows you to interact with [SOAP](https://en.wikipedia.org/wiki/SOAP) service endpoints
```
Comment (christinaausley) [inline docs/components/connectors/protocol/soap.md:15]: Is this the best link we have for info on SOAP? IBM and Oracle have some docs on this too, but Wikipedia might be the best we can do for now.

**Review by christinaausley — DISMISSED** [review]

**Comment by sbuettner** [comment]
Invited to have another pair of eyes from the OG @jonathanlukas

File: docs/components/connectors/protocol/soap.md [hunk 13]
```diff
@@ -0,0 +1,212 @@
+---
+id: soap
+title: SOAP Connector
+sidebar_label: SOAP Connector
+description: The SOAP Connector allows you to connect your BPMN process with SOAP services.
+---
+
+:::note
+The **SOAP Connector** is only supported by Self-Managed Camunda 8 instances.
```
Comment (jonathanlukas) [inline docs/components/connectors/protocol/soap.md:9]: It is also supported in hybrid mode

File: docs/components/connectors/protocol/soap.md [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+:::note
+The **SOAP Connector** currently supports only `SHA-1` password encoding.
+:::
+
+### WSS signature
+
+Use the **WSS signature** in the **Authentication** dropdown when the requested SOAP endpoint requires a message to be
+cryptographically signed with a [signature](http://docs.oasis-open.org/wss-m/wss/v1.1.1/cs01/wss-SOAPMessageSecurity-v1.1.1-cs01.html#_Toc307407954).
+
+Enter all necessary fields according to your service specification.
```
Comment (jonathanlukas) [inline docs/components/connectors/protocol/soap.md:56]: We should be prepared to extend here based on customer feedback :P

File: docs/components/connectors/protocol/soap.md [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+## SOAP message
+
+### SOAP version
+
+Select the latest supported version of the SOAP service.
+
+### SOAPAction HTTP header
+
+Enter the SOAPAction HTTP header that will be used in the request. Leave this value blank if the SOAPAction HTTP header
+won't be used in your request.
```
Comment (jonathanlukas) [inline docs/components/connectors/protocol/soap.md:67]: This is only required in SOAP 1.1

File: docs/components/connectors/protocol/soap.md [hunk 16]
```diff
@@ -0,0 +1,212 @@
...
+Use the **WSS signature** in the **Authentication** dropdown when the requested SOAP endpoint requires a message to be
+cryptographically signed with a [signature](http://docs.oasis-open.org/wss-m/wss/v1.1.1/cs01/wss-SOAPMessageSecurity-v1.1.1-cs01.html#_Toc307407954).
+
+Enter all necessary fields according to your service specification.
+
+## SOAP message
+
+### SOAP version
+
+Select the latest supported version of the SOAP service.
```
Comment (jonathanlukas) [inline docs/components/connectors/protocol/soap.md:62]:
```suggestion
Select the desired version of the SOAP service.
```

File: docs/components/connectors/protocol/soap.md [hunk 17]
```diff
@@ -0,0 +1,212 @@
...
+
+When **Template** is chosen, enter the **XML template** value, for example `<camunda:Param><camunda:ParamType>{{paramValue}}</camunda:ParamType></camunda:Param>`.
+
+Enter the **XML template context** value, for example `={paramValue: 1234567890}`, and enter the **Namespaces** value, for example `={"camunda":"http://my.service.com/webservicesserver/"}`.
+
+#### XML compatible JSON
+
+When **XML compatible JSON** is chosen, enter the **JSON definition**, for example
+
+```xml
```
Comment (jonathanlukas) [inline docs/components/connectors/protocol/soap.md:88]: This is XML, I would have expected a json document here, plus its "transformed" xml equivalent.

File: docs/components/connectors/protocol/soap.md [hunk 18]
```diff
@@ -0,0 +1,212 @@
...
+
+Enter the **Namespaces** value, for example `={"camunda":"http://my.service.com/webservicesserver/"}`.
+
+## Output mapping
+
+### Result variable
+
+You can export a complete response from a SOAP call into a dedicated variable accessible anywhere in a process.
+To do so, input a variable name in the **Result variable** field. Use a unique name to avoid
+overwriting variables.
```
Comment (jonathanlukas) [inline docs/components/connectors/protocol/soap.md:102]: Here should add the schema of the output here:
```json
{
  "response": {
    "Envelope": {
      "Header": {
        <header content>
      },
      "Body": {
        <body content>
      } 
    }
  }
}
```

File: docs/components/connectors/protocol/soap.md [hunk 19]
```diff
@@ -0,0 +1,212 @@
...
+
+## Usage examples
```
Comment (jonathanlukas) [inline docs/components/connectors/protocol/soap.md:104]: Can we have an example with response mapping as well or is this unusual?

**Review by jonathanlukas — DISMISSED** [review]

File: docs/components/connectors/protocol/soap.md (hunk 12 above)
Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md:15]: AFAIK, IBM and Oracle have their sort of "proprietary" flavors. If wikipedia is not reliable source here, maybe consider w3c https://www.w3.org/TR/soap/?

File: docs/components/connectors/protocol/soap.md (hunk 13 above)
Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md]: Updated

File: docs/components/connectors/protocol/soap.md (hunk 14 above)
Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md:56]: I don't have a valid example. Do you mind adding one?

File: docs/components/connectors/protocol/soap.md (hunk 16 above)
Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md]: Updated

File: docs/components/connectors/protocol/soap.md (hunk 15 above)
Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md]: Updated

File: docs/components/connectors/protocol/soap.md (hunk 18 above)
Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md:104]: Updated this section.

File: docs/components/connectors/protocol/soap.md (hunk 19 above)
Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md:152]: Added this section

File: docs/components/connectors/protocol/soap.md (hunk 14 above)
Comment (akeller) [reply docs/components/connectors/protocol/soap.md:56]: @jonathanlukas, do you have something already? Or can we resolve this?

Comment (jonathanlukas) [reply docs/components/connectors/protocol/soap.md:56]: not yet

File: docs/components/connectors/protocol/soap.md [hunk 20]
```diff
@@ -0,0 +1,260 @@
...
+
+```json
+= {
+  "camunda:Object01": {
+    "camunda:Object02": myObjectValue
+  }
+}
+```
+
+Enter the **Namespaces** value, for example `={"camunda":"http://my.service.com/webservicesserver/"}`.
```
Comment (jonathanlukas) [inline docs/components/connectors/protocol/soap.md:96]:
```suggestion
#### Namespaces

All namespaces required to create a valid SOAP request should be added here.

Enter the **Namespaces** value, for example `={"camunda":"http://my.service.com/webservicesserver/"}`.
```

File: docs/components/connectors/protocol/soap.md [hunk 21]
```diff
@@ -0,0 +1,260 @@
+---
+id: soap
+title: SOAP Connector
+sidebar_label: SOAP Connector
+description: The SOAP Connector allows you to connect your BPMN process with SOAP services.
+---
+
+:::note
+The **SOAP Connector** is only supported by Self-Managed and Hybrid Camunda 8 instances.
```
Comment (christinaausley) [inline docs/components/connectors/protocol/soap.md:9]: What do we mean when we say "Hybrid Camunda 8"? Is this the same as Self-Managed?

Comment (jonathanlukas) [reply docs/components/connectors/protocol/soap.md:9]: Hybrid means that you are running the cluster in Saas, but your connector runtime is local.

Comment (christinaausley) [reply docs/components/connectors/protocol/soap.md:9]: @akeller Do we have this defined anywhere, and I'm just missing it? If not, can add to glossary if this is a term we may use more often.

Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md:9]: @christinaausley we added it here previously: https://docs.camunda.io/docs/guides/use-connectors-in-hybrid-mode/

Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md:9]: I mean, introduced the term

Comment (christinaausley) [reply docs/components/connectors/protocol/soap.md:9]: Ah! Got it. I may open an issue in the backlog to link to this from within the Connector docs and define this in our glossary. My mind went to badges, and wondering how we would flag things moving forward -- in this case, we can just lowercase hybrid and link to the hybrid mode guide when referenced in the Connector docs. WDYT?

Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md:9]: Works for me

Comment (akeller) [reply docs/components/connectors/protocol/soap.md:9]: To my knowledge, Connectors is the only feature used in hybrid mode, but it would be great to have it in the glossary @christinaausley!

File: docs/components/connectors/protocol/soap.md (hunk 12 above)
Comment (akeller) [reply docs/components/connectors/protocol/soap.md]: I'm good with either external link. Since SOAP is a protocol, everyone wants their own flavor 🍦

Comment (igpetrov) [reply docs/components/connectors/protocol/soap.md]: I added the `https://www.w3.org/TR/soap/` link. I believe this one is the most generic one

**Comment by christinaausley** [comment]
@igpetrov Any additional changes here? Would you like this merged for release tomorrow?

**Comment by igpetrov** [comment]
Hi @christinaausley 

No more changes here. Let's go ahead and merge this. Thank you!

**Review by christinaausley — APPROVED** [review]
